server/bin/pbench-server-prep-shim-002.py

```diff
@@ -110,147 +79,41 @@ def process_tb(config, logger, receive_dir, qdir_md5, duplicates, errors):
         os.path.join(receive_dir, "**", "*.tar.xz.md5"), recursive=True
     )
 
-    archive = config.ARCHIVE
     logger.info("{}", config.TS)
     list_check.sort()
     nstatus = ""
 
-    ntotal = ntbs = nerrs = nquarantined = ndups = 0
+    ntotal = ntbs = nerrs = 0
+
+    # FIXME: keep a generated token which has longer duration
```

Comment:
Yeah, this is an interesting detail, as we don't yet actually have support for a "long[er] duration" token, and we'll need one to make this new "push shim" workable. We'll also need a mechanism to create the server account to which we'll be pushing, whether we call it "pbench" or "public" or something else.

Reply:
As far as I'm concerned, the "push shim" is more or less unique (i.e., I expect that there will only be one per server, and there may only be one deployed _ever_).  Therefore, I don't think we need to worry _too_ much about a lot of mechanism here for it.  All that we really need is a token value which won't expire too soon, and I think we could produce that more or less manually, if necessary (assuming that the existing mechanism in the server can handle it when it receives a token with a really long duration).

Likewise for the "server account":  I think we just create one (more or less manually), which we'll do when we create the token value.

So, the requirement for the "push shim" is some secure way of storing and retrieving the token so it can use it in the server requests.  Obviously, we shouldn't put it in the code, but it's probably adequate to put it in a config file.  I don't know whether we want to create a separate file just for the "push shim" or whether we want to create a separate section in the one of the existing config files (which would never appear except in a/the shim deployment).

Reply:
Interesting point; yes, the push-shim is a temporary (if important) migration tool for users of the current Pbench production server. We could indeed do all the setup manually, and I think that also argues for not putting the token (and any other unique variables) in `pbench-server-default.cfg` ... there's no point in confusing "real" users and nobody else will ever need to set them.

Reply:
That makes a good point for keeping the server secured. I will get rid of the put-shim cfg file.